sources/packages/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts

```diff
@@ -329,8 +342,34 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
         { coeThresholdDate },
       );
     }
-    coeQuery.orderBy("disbursementSchedule.coeStatus");
-    return coeQuery.getMany();
+    if (paginationOptions.searchCriteria) {
+      coeQuery.andWhere(
+        new Brackets((qb) => {
+          qb.where(
+            "CONCAT(user.firstName,  ' ', user.lastName ) Ilike :searchCriteria",
+            {
+              searchCriteria: `%${paginationOptions.searchCriteria.trim()}%`,
+            },
+          ).orWhere("application.applicationNumber Ilike :searchCriteria", {
+            searchCriteria: `%${paginationOptions.searchCriteria}%`,
+          });
+        }),
+      );
+    }
+    coeQuery
+      .orderBy(
+        this.transformToEntitySortField(
```

Comment:
There is already a way to do the map part introduce by @ann-aot in a previous PR. Can you please check it?

Reply:
Leaving for future discussion as discussed.